src/Symfony/Component/HttpKernel/Controller/ArgumentResolver.php

```diff
@@ -72,6 +72,10 @@ public function getArguments(Request $request, callable $controller, \Reflection
                     }
 
                     $argumentValueResolvers = [$this->namedResolvers->get($resolverName)];
+
+                    if ($this->namedResolvers->has(DefaultValueResolver::class) && !$argumentValueResolvers[0] instanceof DefaultValueResolver) {
```

Comment:
what about this instead?
```suggestion
                    $argumentValueResolvers[] = new DefaultValueResolver();
```

Reply:
I think you would still need to check that `$this->namedResolvers->get($resolverName)` isn't an instance of `DefaultValueResolver`, but other then that I can update it.

Reply:
no need to check for that: if it's the default resolver, then the first will do it's job

Reply:
You're right, if some pinned the `DefaultValueResolver` then it's unlikely that there's no default value. Done.